src/commonTest/kotlin/dev/romainguy/kotlin/math/MatrixTest.kt

```diff
@@ -0,0 +1,399 @@
+    fun mat3_multiplication() {
+        val a = Mat3.of(1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f, 9.0f)
+        val b = Mat3.of(2.0f, 0.0f, 1.0f, 2.0f, 3.0f, 1.0f, 2.0f, 0.0f, 1.0f)
+        assertEquals(
+            Mat3.of(12.0f, 6.0f, 6.0f, 30.0f, 15.0f, 15.0f, 48.0f, 24.0f, 24.0f),
+            a * b,
+        )
+    }
+
+    @Test
+    fun mat4_multiplication() {
+        val a = Mat4.of(
+             1.0f,  2.0f,  3.0f,  4.0f,
+             5.0f,  6.0f,  7.0f,  8.0f,
+             9.0f, 10.0f, 11.0f, 12.0f,
+            13.0f, 14.0f, 15.0f, 16.0f,
+        )
+        val b = Mat4.of(
+            2.0f, 0.0f, 1.0f, 2.0f,
+            1.0f, 1.0f, 2.0f, 0.0f,
+            2.0f, 1.0f, 2.0f, 2.0f,
+            0.0f, 1.0f, 1.0f, 2.0f,
+        )
+        assertEquals(
+            Mat4.of(
+                10.0f,  9.0f, 15.0f, 16.0f,
+                30.0f, 21.0f, 39.0f, 40.0f,
+                50.0f, 33.0f, 63.0f, 64.0f,
+                70.0f, 45.0f, 87.0f, 88.0f,
+            ),
+            a * b
+        )
+    }
+
+    @Test
+    fun mat3_identity() {
+        assertEquals(
+            Mat3(
+                Float3(1f, 0f, 0f),
+                Float3(0f, 1f, 0f),
+                Float3(0f, 0f, 1f),
+            ),
+            Mat3.identity()
+        )
+    }
+
+    @Test
+    fun mat3_of_fails_if_less_than_9_arguments() {
+        assertFailsWith<IllegalArgumentException> {
+            Mat3.of(*8.floatArray())
+        }
+    }
+
+    @Test
+    fun mat3_of() {
+        assertEquals(MAT_3, Mat3.of(*9.floatArray()))
+    }
+
+    @Test
+    fun mat4_identity() {
+        assertEquals(
+            Mat4(
+                Float4(1f, 0f, 0f, 0f),
+                Float4(0f, 1f, 0f, 0f),
+                Float4(0f, 0f, 1f, 0f),
+                Float4(0f, 0f, 0f, 1f),
+            ),
+            Mat4.identity()
+        )
+    }
+
+    @Test
+    fun mat4_of_fails_if_less_than_16_arguments() {
+        assertFailsWith<IllegalArgumentException> {
+            Mat4.of(*15.floatArray())
+        }
+    }
+
+    @Test
+    fun mat4_of() {
+        assertEquals(MAT_4, Mat4.of(*16.floatArray()))
+    }
+
+    @Test
+    fun transpose_mat3() {
+        assertEquals(
+            Mat3(
+                Float3(1f, 2f, 3f),
+                Float3(4f, 5f, 6f),
+                Float3(7f, 8f, 9f)
+            ),
+            transpose(MAT_3)
+        )
+    }
+
+    @Test
+    fun transpose_Mat3_of_identity_is_identity() {
+        assertEquals(transpose(Mat3.identity()), Mat3.identity())
+    }
+
+    @Test
+    fun inverse_mat3() {
+        assertEquals(
+            Mat3(
+                Float3(0f, 1f, 0f),
+                Float3(-2f, 1f, 1f),
+                Float3(2f, -2f, 0f)
+            ),
+            inverse(Mat3(
+                Float3(1f, 0f, 0.5f),
+                Float3(1f, 0f, 0f),
+                Float3(1f, 1f, 1f)
+            ))
+        )
+    }
+
+    @Test
+    fun inverse_mat3_of_identity_is_identity() {
```

Comment:
Please don't use _. If backticks can't be used, use camel case instead: `inverseMat3OfIdentityIsIdentity`

Reply:
Changed to camel case. Originally, since the identifiers were more like a description, I wanted to preserve clear spaces.